### DISCUSSION TITLE
스쿼시 머지 vs 리베이스 머지

### DISCUSSION THREAD

**Discussion by ljw0096** [prgrms-fe-devcourse/FEDC2_TimeDivider_Root#5] [1. 비개발]
## 스쿼시

- PR 날린 내부의 Commit Log가 main branch에 남지않음
- 따라서 History 관리를  PR에 적은 내용을 기반으로 추적할 수 있게 merge commit에 태그를 남겨줌

## 리베이스
- 머지커밋이 생성되지 않음
- PR날린 브랜치 내부의 커밋 로그가 그대로 Merge

### Issue
- 스쿼시를 사용하기로 결정해서 Commit convention을 너무 Strict하게 정하지 않아도 된다?
- No. 개개인이 PR 내부에서의 커밋을 확인할 수 있어서  컨벤션을 잘 정리해두는게 좋을 것 같다

**Comment by dahye1013** [comment]
PR에서 리뷰할때 커밋 아이디 기준으로 어노테이션을 달아서 할 수 있으니까, 커밋명 자체는 명시적으로 하는게 좋다고 생각해요! 
Commit convention 을 엄격하게 지키기보다는 서로가 소통하기 용이하도록! 이것도 비효율이 느껴진다면 또 토론해서 바꿔보죠 💪
